tests/test_shared/test_shared.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+from pathlib import Path
+
+from bioimageio.spec.shared.raw_nodes import URI
+
+
+@pytest.mark.parametrize(
+    "src",
+    [
+        "https://raw.githubusercontent.com/bioimage-io/spec-bioimage-io/main/example_specs/models/unet2d_nuclei_broad/rdf.yaml",
+        Path(__file__),
+        __file__,
+        URI(
+            "https://raw.githubusercontent.com/bioimage-io/spec-bioimage-io/main/example_specs/models/unet2d_nuclei_broad/rdf.yaml"
+        ),
+    ],
+)
+def test_get_resolved_source_path(src):
+    from bioimageio.spec.shared import get_resolved_source_path
+
+    res = get_resolved_source_path(src, root_path=Path(__file__).parent)
+    assert isinstance(res, Path)
+    assert res.exists()
```

Comment:
no way to know _what_ path to expect (in terms of where it's pointing to)?

Reply:
of course for the paths given this is known. 
Also for the url the cache location is known...
but it doesn't really matter.. I don't want to change this test when we change caching.
I'd say if you use this function you don't really care where the path is, but you care that it is a local file path and not a URL

Reply:
One could mock caching though, to make the test independent - in the end you don't want to test caching here...

Reply:
hmm.. going a bit overboard here I think, but ok...